napari/layers/points/points.py

```diff
@@ -806,19 +796,38 @@ def current_symbol(self, symbol: Union[None, float]) -> None:
 
     @property
     def size(self) -> np.ndarray:
-        """(N, D) array: size of all N points in D dimensions."""
+        """(N,) array: size of all N points."""
         return self._size
 
     @size.setter
     def size(self, size: Union[int, float, np.ndarray, list]) -> None:
         try:
-            self._size = np.broadcast_to(size, self.data.shape).copy()
+            self._size = np.broadcast_to(size, len(self.data)).copy()
         except ValueError as e:
-            try:
-                self._size = np.broadcast_to(
-                    size, self.data.shape[::-1]
-                ).T.copy()
-            except ValueError:
+            if (
+                'input operand has more dimensions than allowed by the axis remapping'
+                in e.args
+            ):
+                try:
+                    self._size = np.broadcast_to(
+                        np.array(size)[:, -1], len(self.data)
```

Comment:
The last column seems as arbitrary as the mean one, but the mean one (while still arbitrary) has the advantage of effectively maintaining the old behavior for visualization (in case anyone was actually relying on that). In other words, if we're deprecating this behavior, why change it too?

Reply:
Good point!